ui/test/nuxeo-aggregation-widget.test.js

```diff
@@ -0,0 +1,364 @@
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} sort-by-label></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(9);
+      expect(checkboxes[0].textContent.trim()).to.equal('Asimov (1)');
+      expect(checkboxes[1].textContent.trim()).to.equal('Balzac (7)');
+      expect(checkboxes[2].textContent.trim()).to.equal('Cervantes (8)');
+    });
+  });
+
+  suite('Collapsible', () => {
+    test('Should display a label in a button when "label" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible label="Some label"> </nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      const label = nuxeoCollapsible.shadowRoot.querySelectorAll('slot')[0].assignedNodes()[0];
+      expect(label.innerHTML).to.equal('Some label');
+    });
+
+    test('Should display a collapsible element and hide checkboxes when "collapsible" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      expect(isElementVisible(nuxeoCollapsible)).to.be.true;
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+    });
+
+    test('Should display checkboxes when "collapsible" is set and it is tapped', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      const nuxeoCollapsibleButton = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(8);
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+
+      tap(nuxeoCollapsibleButton);
+      expect(checkboxes.length).to.equal(8);
+      for (let i = 0; checkboxes.length <= 7; i++) {
+        expect(isElementVisible(checkboxes[i])).to.be.true;
+      }
+      expect(isElementVisible(checkboxes[8])).to.be.false;
```

Comment:
If the length is 8 why do we have `checkboxes[8]` entry? 🤔

Reply:
Since the default visible items is 8, the ninth checkbox should be invisible, hence checkbox[8] visibility should be false.